includes/admin/extensions/class-extensions-api.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+namespace EDD\Admin\Extensions;
+
+class ExtensionsAPI {
+
+	/**
+	 * Gets the product data from the EDD Products API.
+	 *
+	 * @since 2.11.x
+	 * @param array $body    The body for the API request.
+	 * @param int   $item_id The product ID, if querying a single product.
+	 * @return object
+	 */
+	public function get_product_data( $body = array(), $item_id = false ) {
+		if ( empty( $body ) ) {
+			if ( empty( $item_id ) ) {
+				return false;
+			}
+			$body = $this->get_api_body( $item_id );
+		}
+		$key         = $this->array_key_first( $body );
+		$option_name = "edd_extension_{$key}_{$body[ $key ]}_data";
```

Comment:
Is there any risk here of the `$body` causing an option name that is too long? Should we MD5 this key so that there is less risk of a large option name?

Reply:
At this point, I don't _think_ so, as the values will be:
- `$key`: `product`, `category`, or `tag`
- `$body[ $key ]`: a product or term ID (integer)

![image](https://user-images.githubusercontent.com/4853066/143921674-4e25b3f3-a595-4ca5-a4a2-2a14a36c7910.png)

I'm willing to hash it if that's safest but for right now I confess it's been helpful to have a recognizable option name.

Reply:
> I confess it's been helpful to have a recognizable option name.

I like that too.

Reply:
I do like it as well, just want to make sure we keep that in mind as we use this in the future. :+1: Can we make a comment or something about how the key is created so we don't overrun it by accident?